core/metrics/metrics.go

```diff
@@ -43,8 +43,14 @@ var (
 	DefaultReporterInterval = time.Second
 )
 
-// ReporterFunc is used during service init time to register the reporter
-type ReporterFunc func(config.ConfigurationProvider) (tally.StatsReporter, error)
+// ScopeInit interface provides necessary data to properly initialize root metrics scope
```

Comment:
maybe mention that service.Host conforms to this interface and it's here to avoid an import cycle